cmd/syncProcessor.go

```diff
@@ -175,9 +176,32 @@ type localFileDeleter struct {
 	rootPath string
 }
 
+// As at version 10.4.0, we intentionally don't delete directories in sync,
+// even if our folder properties option suggests we should.
+// Why? The key difficulties are as follows, and its the third one that we don't currently have a solution for.
+// 1. Timing (solvable in theory with FolderDeletionManager)
+// 2. Identifying which should be removed when source does not have concept of folders (e.g. BLob)
+//    Probably solution is to just respect the folder properties option setting (which we already do in our delete processors)
+// 3. In Azure Files case (and to a lesser extent on local disks) users may have ACLS or other properties
```

Comment:
Remember that hdi_isfolder blobs exist. If we're respecting the folder properties option, we should perhaps respect these folder-esque representations.

The other problem with this is that we're trying to sync from something that's objectively not a filesystem to something that objectively is a filesystem.

Reply:
Yes. We have a separate backlog item for the hdi_isfolder case.